This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Thu, 28 Mar 2024 18:29:25 +0100
Subject: [PATCH 1/2] Ignore any errors from native-static-libs query

It will not properly work with `-Zbuild-std`.

diff --git a/meson.build b/meson.build
index 1111111..2222222 100644
--- a/meson.build
+++ b/meson.build
@@ -306,13 +306,8 @@ if host_system == 'windows'
   native_libs = run_command(
     rustc_query_native_static_libs_args,
     capture: true,
-    check: true
   )
 
-  if native_libs.stderr() != ''
-    error(native_libs.stderr())
-  endif
-
   foreach i: native_libs.stdout().split()
     if i != 'msvcrt'
       private_dependencies += cc.find_library(i, required: get_option('default_library') != 'shared')

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Fri, 7 Jul 2023 18:41:29 +0200
Subject: [PATCH 2/2] Disable PDF/PostScript features in cairo-rs

We build Cairo with `-Dzlib=disabled`, which implicitly disables
these surface backends.

diff --git a/librsvg-c/Cargo.toml b/librsvg-c/Cargo.toml
index 1111111..2222222 100644
--- a/librsvg-c/Cargo.toml
+++ b/librsvg-c/Cargo.toml
@@ -10,7 +10,7 @@ edition.workspace = true
 rust-version.workspace = true
 
 [dependencies]
-cairo-rs = { version = "0.19", features=["v1_16", "png", "pdf", "ps", "svg"] }
+cairo-rs = { version = "0.19", features=["v1_16", "png", "svg"] }
 cast = "0.3.0"
 float-cmp = "0.9.0"
 gdk-pixbuf = "0.19"
diff --git a/rsvg/Cargo.toml b/rsvg/Cargo.toml
index 1111111..2222222 100644
--- a/rsvg/Cargo.toml
+++ b/rsvg/Cargo.toml
@@ -47,7 +47,7 @@ name = "rsvg"
 [dependencies]
 # Keep these in sync with respect to the cairo-rs version:
 #   src/lib.rs - toplevel example in the docs
-cairo-rs = { version = "0.19", features=["v1_16", "png", "pdf", "ps", "svg"] }
+cairo-rs = { version = "0.19", features=["v1_16", "png", "svg"] }
 cast = "0.3.0"
 cssparser = "~0.31"
 data-url = "0.3.0"
